core/src/main/java/io/kestra/core/models/tasks/runners/ScriptService.java

```diff
@@ -59,41 +57,37 @@ public static String replaceInternalStorage(
             .replaceAll(throwFunction(matchResult -> {
                 String localFile = saveOnLocalStorage(runContext, matchResult.group()).replace("\\", "/");
 
-                internalStorageToLocalFileConsumer.accept(matchResult.group(), localFile);
-
                 if (!replaceWithRelativePath) {
                     return localFile;
                 }
 
-                return localFile.startsWith("/") ? localFile.substring(1) : localFile;
+                return runContext.workingDir().path().relativize(Path.of(localFile)).toString();
```

Comment:
There is a relativize method on the runContext.workingDir() (or directly on runContext, not sure since the refacto) that protect against path traversal, please use it instead.